File: app/src/main/java/autogravity/com/mvvmdemo/domain/schedulers/ImmediateSchedulerProvider.java
package autogravity.com.mvvmdemo.domain.schedulers;

import android.support.annotation.NonNull;
import io.reactivex.Scheduler;
import io.reactivex.schedulers.Schedulers;

/**
 * Created by LeviHsiao on 4/10/18.
 */

public class ImmediateSchedulerProvider implements BaseSchedulerProvider {

    @NonNull @Override public Scheduler computation() {
        return Schedulers.trampoline();
    }

    @NonNull @Override public Scheduler io() {
        return Schedulers.trampoline();
    }

    @NonNull @Override public Scheduler mainThread() {
        return Schedulers.trampoline();
    }

    @NonNull @Override public Scheduler newThread() {
        return Schedulers.trampoline();
    }
}

File: app/src/main/java/autogravity/com/mvvmdemo/domain/schedulers/ImmediateSchedulerProvider.kt
package autogravity.com.mvvmdemo.domain.schedulers

import io.reactivex.Scheduler
import io.reactivex.schedulers.Schedulers

/**
 * Created by LeviHsiao on 4/10/18.
 */

class ImmediateSchedulerProvider : BaseSchedulerProvider {

    override fun computation(): Scheduler {
        return Schedulers.trampoline()
    }

    override fun io(): Scheduler {
        return Schedulers.trampoline()
    }

    override fun mainThread(): Scheduler {
        return Schedulers.trampoline()
    }

    override fun newThread(): Scheduler {
        return Schedulers.trampoline()
    }
}
